src/operator.rs

```diff
@@ -119,13 +136,27 @@ pub fn apply_or(l: Num, r: Num) -> Option<Num> {
     Some(if l != 0 { l } else { r })
 }
 pub fn apply_or_logical(l: Num, r: Num) -> Option<Num> {
-    Some((l != 0 || r != 0) as Num)
+    Some(
+        (l != 0
+            || if Some(r) == ERROR_VALUE {
+                return ERROR_VALUE;
+            } else {
+                r != 0
+            }) as Num,
+    )
```

Comment:
Does something like this work?
```suggestion
    match (l, r) {
        (0, r) if Some(r) == ERROR_VALUE => ERROR_VALUE,
        (l, r) => Some((l != 0 || r != 0) as Num),
    }
```